### PR TITLE
Completed Scrabble project at breakfast level w/ some lunch

### PR DESCRIPTION
**Pull request by shannachau**
This project is completed at the breakfast level. I also was giving part of lunch level a try. For the lunch level, I completed the following: 
- self.new(name): creates a new instance with the instance variable name assigned
- #name: returns the @name instance variable
- #plays: returns an Array of the words played by the player
- #play(word): Adds the input word to the plays Array

For the breakfast level, I was trying to implement custom error messages using raise, but I was having a hard time figuring out the syntax for the rspec test. I ran out of time, so I left out a lot of edge case testing. :( If you have a suggestion for how to do that or if you could even refer me to a good walkthrough/example that would be great!! Going to try and figure it out tomorrow/get some help on that! 

I would love feedback on my self.highest_score_from method. It works, but I feel like there might have been a simpler way to implement it. Also would like feedback for the rspec tests I wrote. I wasn't completely sure if I was organizing them correctly or adding in the right descriptions. 

Thanks for reviewing!!

### CODE REVIEW

File: lib/scrabble.rb [hunk 1]
```diff
@@ -1,4 +1,104 @@
...
+    def self.highest_score_from(array_of_words)
+      # create hash using given array
+      # ["cat", "dog"] => {total_points: "cat", total_points: "dog"}
+      word_points_hash = array_of_words.group_by {|word| score(word)}
+
+      # sorts through the keys in the hash to find largest point value
+      # returns an array of word(s) that have the largest point value
+      # .max_by returns an array like so: [4, ["cat", "dog"]]
+      # need to use index [1] to access just the words
+      highest_scoring_words = word_points_hash.max_by {|k, v| k }[1]
```
Comment (kerrizor) [inline lib/scrabble.rb:58]: rather than `[]` you could do `.last` :)

File: lib/scrabble.rb [hunk 2]
```diff
@@ -1,4 +1,104 @@
...
+          end
+        end
+
+        # otherwise, return the word in the array with the least amount of letters
+        # this also returns the first instance in the array with the least amount of letters
+        return highest_scoring_words.min_by {|word| word.length}
+
+      # if the array does not contain more than one element, just return the first/only element
+      else
+        highest_scoring_words[0]
```
Comment (kerrizor) [inline lib/scrabble.rb:77]: `.first`

**Comment by shannachau** [comment]
Hey there! Thank you for your comments. Really like the idea of using .first and .last. Definitely helps clarify the code. Would love some feedback on my rspec tests if you have the time! Thank you!
